src/main/java/com/mongodb/kafka/connect/sink/processor/TimeseriesTimeFieldAutoConversion.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.sink.processor;
+
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.TIMESERIES_TIMEFIELD_AUTO_CONVERSION_DATE_FORMAT_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.TIMESERIES_TIMEFIELD_CONFIG;
+import static java.lang.String.format;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.ResolverStyle;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.bson.BsonDateTime;
+import org.bson.BsonValue;
+
+import com.mongodb.kafka.connect.sink.MongoSinkTopicConfig;
+import com.mongodb.kafka.connect.sink.converter.SinkDocument;
+
+class TimeseriesTimeFieldAutoConversion extends PostProcessor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PostProcessor.class);
+
+  private final String fieldName;
+  private final DateTimeFormatter formatter;
+
+  TimeseriesTimeFieldAutoConversion(final MongoSinkTopicConfig config) {
+    super(config);
+    fieldName = config.getString(TIMESERIES_TIMEFIELD_CONFIG);
+    formatter =
+        DateTimeFormatter.ofPattern(
+                config.getString(TIMESERIES_TIMEFIELD_AUTO_CONVERSION_DATE_FORMAT_CONFIG))
+            .withResolverStyle(ResolverStyle.LENIENT);
```

Comment:
I've updated and no longer specify a resolver style. As this is user based we will need to accommodate `yyyy`  because it will be what users are used to.